lib/pure/asyncdispatch.nim

```diff
@@ -1101,12 +1134,34 @@ else:
 
     AsyncEvent* = distinct SelectEvent
 
+    VirtualFD* = distinct int
+
+    VirtualAsyncEventImpl = object
+      triggered: bool
+      when compileOption("threads"):
+        eventLock: RLock
+      p: PDispatcher
+      vFD: VirtualFD
+      cb: Callback
+    VirtualAsyncEvent* = ptr VirtualAsyncEventImpl
+
```

Comment:
These are defined twice. Can you reduce the duplication?

Reply:
Based on your comment below about `deInitVirtualEventDispatcher` I think I can clean this up, but I think the type dependency loop will still be there. I can't figure out how to get rid of that.

Reply:
@dom96,
Sorry :/
This is a terrible and convoluted explanation. Hit me up on irc if you can. I might be able to explain it better by walking through it with you live.

Reply:
@Araq, I haven't been able to catch you on irc the last few days. But I would like to get your thoughts on this type dependency nonsense. I don't see a way around it, but maybe an expert(TM) can help :-P

Reply:
I talked to @Araq briefly about this on IRC. He didn't look too hard, but generally this kind of thing will happen in x-platform situations until we get recursive module definitions. So it's a necessary evil for now.